service.py

```diff
@@ -0,0 +1,58 @@
+from db import Session
+from models import Employee, Department
+from sqlalchemy.orm.exc import NoResultFound
+
+class EmployeeService:
+    def __init__(self):
+        self.session = Session()
+
+    def add_employee(self, first_name, last_name, salary, department_name, hire_date):
+        department = self.session.query(Department).filter_by(name=department_name).first()
+        if not department:
+            department = Department(name=department_name)
+            self.session.add(department)
+            self.session.commit()
+        employee = Employee(first_name=first_name, last_name=last_name, salary=salary,
+                            department=department, hire_date=hire_date)
+        self.session.add(employee)
+        self.session.commit()
+        return employee
+
+    def update_employee(self, employee_id, **kwargs):
+        employee = self.session.query(Employee).filter_by(id=employee_id).first()
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider reusing existing employee query method</b></div>
<div id="fix">

Consider using `get_employee()` helper method instead of duplicating the query logic in multiple places. The same query pattern `self.session.query(Employee).filter_by(id=employee_id).first()` is repeated.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -21,8 +21,7 @@
     def update_employee(self, employee_id, **kwargs):
 -        employee = self.session.query(Employee).filter_by(id=employee_id).first()
 +        employee = self.get_employee(employee_id)
          if not employee:
              raise NoResultFound(f"Employee with id {employee_id} does not exist.")

 @@ -31,8 +30,7 @@
     def delete_employee(self, employee_id):
 -        employee = self.session.query(Employee).filter_by(id=employee_id).first()
 +        employee = self.get_employee(employee_id)
          if not employee:
              raise NoResultFound(f"Employee with id {employee_id} does not exist.")
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/rishabhbohra3/Bito-CRA-Issues-Dataset/pull/12#issuecomment-2514850927>#ad7e48</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged